components/ai_chat/core/browser/conversation_handler_unittest.cc

```diff
@@ -2239,4 +2256,185 @@ TEST_P(PageContentRefineTest, TextEmbedderInitialized) {
   }
 }
 
+TEST_F(ConversationHandlerUnitTest, RateMessage) {
+  // Create a sample chat history with 2 turns (human, assistant)
+  std::vector<mojom::ConversationTurnPtr> history = CreateSampleChatHistory(1);
+  ASSERT_EQ(history.size(), 2u);
+
+  // Store UUIDs for easy access
+  const std::string human_turn_uuid = history[0]->uuid.value();
+  const std::string assistant_turn_uuid = history[1]->uuid.value();
+
+  // Initialize the conversation handler with test history
+  conversation_handler_->SetChatHistoryForTesting(CloneHistory(history));
+
+  // Test when model_key is null (should use current model)
+  {
+    // Clear the model_key on the assistant turn
+    conversation_handler_->GetConversationHistory().back()->model_key =
+        std::nullopt;
+
+    // Set expectations for the mock
+    std::optional<std::string> received_rating_id;
+
+    // Should use the current model from GetCurrentModel()
+    EXPECT_CALL(*mock_feedback_api_,
+                SendRating(true, false, _, "automatic", _, _))
+        .WillOnce(
+            [&](bool is_liked, bool is_premium,
+                const base::span<const mojom::ConversationTurnPtr>&
+                    history_span,
+                const std::string& model_name,
+                const std::string& selected_language,
+                api_request_helper::APIRequestHelper::ResultCallback callback) {
+              // Verify the history being sent contains the human and assistant
+              // turns
+              EXPECT_EQ(history_span.size(), 2u);
+
+              // Create a mock response with an ID
+              base::Value::Dict response_dict;
+              response_dict.Set("id", "test-rating-current-model");
+              base::Value response(std::move(response_dict));
+
+              // Return the response via callback
+              std::move(callback).Run(api_request_helper::APIRequestResult(
+                  200,                  // response_code
+                  std::move(response),  // value_body
+                  {},                   // empty headers
+                  net::OK,              // error_code
+                  GURL()));             // empty final_url
+            });
+
+    // Call RateMessage
+    base::RunLoop run_loop;
+    conversation_handler_->RateMessage(
+        true, assistant_turn_uuid,
+        base::BindLambdaForTesting(
+            [&](const std::optional<std::string>& rating_id) {
+              received_rating_id = rating_id;
+              run_loop.Quit();
+            }));
+    run_loop.Run();
```

Comment:
we can use `base::test::TestFuture` and get rid of RunLoop in all newly added occurrences in this file

Reply:
https://github.com/brave/brave-core/pull/28999/commits/2ecb95ee89e21b487944564beba1f040d0a11752